lib/langchain/assistants/assistant.rb

```diff
@@ -183,6 +133,128 @@ def instructions=(new_instructions)
 
     private
 
+    # Check if the run is finished
+    #
+    # @param state [Symbol] The current state
+    # @return [Boolean] Whether the run is finished
+    def run_finished?(state)
+      [:completed, :failed, :expired].include?(state)
+    end
+
+    # Handle the current state and transition to the next state
+    #
+    # @param state [Symbol] The current state
+    # @param auto_tool_execution [Boolean] Whether or not to automatically run tools
+    # @return [Symbol] The next state
+    def handle_state(state, auto_tool_execution)
+      case state
+      when :in_progress
+        process_latest_message(auto_tool_execution)
+      when :running_tools
```

Comment:
I'm wondering if we should just copy how OpenAI calls it and name it `:requires_action` to be more aligned with that interface?

Reply:
Yes, sure, I agree!